spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -438,6 +460,7 @@ private Channel doCreateBareChannel(ChannelCachingConnectionProxy connection, bo
 
 	@Override
 	public final Connection createConnection() throws AmqpException {
+		Assert.state(!this.stopped, "The ConnectionFactory isn't active any more for obtaining connection.");
```

Comment:
`"The ApplicationContext is closed and the ConnectionFactory can no longer create connections."`

I was going to merge with this minor change, but it appears we might have introduced some instability...

`Class org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainerIntegrationTests`

`testListenerSunnyDay[7]`

>java.lang.AssertionError: Timed out waiting for message
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainerIntegrationTests.doSunnyDayTest(SimpleMessageListenerContainerIntegrationTests.java:215)
	at org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainerIntegrationTests.testListenerSunnyDay(SimpleMessageListenerContainerIntegrationTests.java:182)